flake8_trio/visitors/flake8triovisitor.py

```diff
@@ -202,6 +196,9 @@ def save_state(self, node: cst.BaseStatement, *attrs: str, copy: bool = False):
         else:
             self.outer[node] = state
 
+    def restore_state(self, node: cst.CSTNode):
+        self.set_state(self.outer.pop(node, {}))
+
```

Comment:
visitor classes are still responsible for restoring the outer state, which is not the case for ast visitors. Might change that at some point, but for now at least added a helper function.